frontend/src/pages/NewStudent.vue

```diff
@@ -0,0 +1,306 @@
+                    <option value = "2">Little (Able to understand simple words)</option>
+                    <option value = "3">Simple (Able to speak full sentences)</option>
+                    <option value = "4">Intermediate (Able to understand simple words)</option>
+                  </b-select>
+              </b-field>  
+
+            </b-field>
+            
+    
+            <b-field label="Notes" class="half-width">
+                <b-input maxlength="200" type="textarea" placeholder="Optional"></b-input>
+            </b-field>
+
+        </section>
+        <b-button class="dark-blue" expanded @click="CreateStudent">Create Student</b-button>
+      </div>
+    </div>
+  </div>      
+</template>
+
+
+<script>
+
+export default {
+  name: 'App',
+
+  data() {
+      return {
+          data: [
+              'Bengali',
+              'English',
+              'Mandarin',
+              'Russian',
+              'Indonesian',
+              'Urdu',
+              'Tamil'
+          ],
+          sourcedata: [
+              'Rotary Club',
+              'Source 1',
+              'Source 2',
+          ],
+          NativeLanguage: '',
+          Source:'',
+          selected: null,
+          file: null,
+      }
+  },
+
+  watch: {
+    file: function(val){
+      this.UploadFile();
+    }
+  },
+
+  computed: {
+    filteredDataArray() {
+        return this.data.filter((option) => {
+            return option
+              .toString()
+              .toLowerCase()
+              .indexOf(this.NativeLanguage.toLowerCase()) >= 0
+            })
+    },
+      filteredsourceDataArray() {
+        return this.sourcedata.filter((sourceOption) => {
+            return sourceOption
+              .toString()
+              .toLowerCase()
+              .indexOf(this.Source.toLowerCase()) >= 0
+            })
+    }  
+  },
+
+
+  methods: {
+    CreateStudent(){
+      this.$buefy.notification.open({
+        message: 'New student added. <u>View profile</u>!',
+        duration: 5000,
+        type: 'is-success',
+        position: 'is-top',
+        // color: '#57A773',
+      })
+    },
+
+    UploadFile(){
+      this.$buefy.notification.open({
+        message: 'The file was uploaded successfully!',
+        duration: 5000,
+        type: 'is-success',
+        position: 'is-top',
+        // color: '#57A773',
+      })
+    },
+
+
+    ErrorUpload(){
+      this.$buefy.notification.open({
+        message: '<b>There was an error in uploading.</b> <br> Please check formatting of the file and try again.',
+        duration: 5000,
+        type: 'is-danger',
+        position: 'is-top',
+        // color: '#57A773',
+      })
+    },
+
+
+
+    showAddLanguage() {
+        this.$buefy.dialog.prompt({
+          message: `Add new language`,
+          inputAttrs: {
+            placeholder: 'e.g. Italian',
+            maxlength: 255,
+            value: this.name
```

Comment:
What is `this.name` referring to here?

Reply:
Refers to the language entered by the user when "Add new language" is selected

Reply:
`this.name` implies there is a `name` property in this component's returned data object. (just like `this.nativeLanguage`, `this.sourceData`...).

the `value` of the buefy dialog prompt should default to user input when not specified. try it out: `this.name` can be removed here without affecting functionality.